Implement blueprint

kylef opened the pull request:
This pull requests adds the initial blueprint for the API Blueprint parsing service adding documentation for Refract parsing.

It uses the hercule tool to transclude the separate JSON/YAML ASTs into the blueprint. It uses the apiary-client to send the blueprint to Apiary on commits to master.

Please review and comment as :+1: to mark this ready for merge, DO NOT merge. I want to merge this myself to check sending it to Apiary works from CircleCI deployment.

pksunkara commented:
I don't understand why we have this PR. Don't you already have apiary.apib in plutonium?

pksunkara commented on source/sample.ast.json:
```diff
@@ -0,0 +1 @@
+{"_version":"2.2","ast":{"_version":"4.0","metadata":[],"name":"","description":"","element":"category","resourceGroups":[{"name":"","description":"","resources":[{"element":"resource","name":"","description":"","uriTemplate":"/message","model":{},"parameters":[],"actions":[{"name":"","description":"","method":"GET","parameters":[],"attributes":{"relation":"","uriTemplate":""},"content":[],"examp ...
```
This file is not indented.

pksunkara commented on source/sample.refract.parse-result.yaml:
```diff
@@ -0,0 +1,89 @@
+                              contentType: text/plain
+                            content: |
+                              Hello World!
+error: null
```
Is this really output from drafter?

kylef replied:
Nope, but it's what the service is actually returning.

pksunkara replied:
We need to fix that. Otherwise, this is not `refract` exactly.

pksunkara commented on source/apiary.apib:
````diff
@@ -0,0 +1,205 @@
+
+```
+application/vnd.refract+json
+application/vnd.refract+yaml
````
Both have `refract.parse-result`

pksunkara commented on source/apiary.apib:
```diff
@@ -0,0 +1,205 @@
+
++ Response 200 (application/vnd.apiblueprint.parseresult+json; version=2.2)
+
+        :[](sample.ast.json)
```
Please name this file, `sample.apiblueprint.parseresult.json`. Similarly, the corresponding `yaml` file.

pksunkara commented on source/apiary.apib:
```diff
@@ -0,0 +1,205 @@
+
++ Request Compose API Blueprint from AST 2.0 sent as YAML (application/vnd.apiblueprint.ast+yaml; version=2.0)
+
+        :[](sample.ast-2.0.yaml)
```
`sample.apiblueprint.ast-2.0.yaml`. Similarly, for `json`.

pksunkara commented:
What's the inconsistency with the source maps? Reviewed.

kylef commented:
@pksunkara thanks for the review, I've addressed your feedback.

As for YAML/JSON source map inconsistencies, looks like these have been resolved in the parser and I've updated them.

pksunkara commented on source/sample.apib:
```diff
@@ -0,0 +1,4 @@
+# GET /message
```
Let's make this the following:

```apib
## /message
### GET
```

That way, we can reuse `sample.apib` in `/compose` part of the blueprint too.

pksunkara commented:
Just one minor comment left.

kylef commented:
@pksunkara addressed.
